flow/project.py

```diff
@@ -3461,20 +3464,32 @@ def _get_submission_operations(
             been collected appropriately from its contained operations.
 
         """
-        for group in self._gather_flow_groups(names):
-            for aggregate in self._get_aggregate_store(group.name).values():
-                if (
-                    group._eligible(aggregate, ignore_conditions)
-                    and self._eligible_for_submission(group, aggregate, cached_status)
-                    and self._is_selected_aggregate(aggregate, aggregates)
-                ):
-                    yield group._create_submission_job_operation(
-                        entrypoint=self._entrypoint,
-                        default_directives=default_directives,
-                        jobs=aggregate,
-                        index=0,
-                        ignore_conditions_on_execution=ignore_conditions_on_execution,
-                    )
+        submission_groups = set(self._gather_flow_groups(names))
+        for (
+            aggregate_store,
+            aggregate_groups,
+        ) in self._group_to_aggregator.inverse.items():
+            matching_groups = set(aggregate_groups) & submission_groups
+            if not matching_groups:
+                # Skip iteration over aggregate store if no groups match the
+                # specified names
+                continue
+            for aggregate in aggregate_store.values():
+                for group in matching_groups:
+                    if (
+                        group._eligible(aggregate, ignore_conditions)
+                        and self._eligible_for_submission(
+                            group, aggregate, cached_status
+                        )
+                        and self._is_selected_aggregate(aggregate, aggregates)
```

Comment:
This is not group-specific so we can consider checking this in the outer loop and continue if it's not the selected aggregate.

Also, I feel like the `_is_selected_aggregate` method is now redundant because we simply just check the availability of `aggregate` in `aggregates`.

Reply:
This is handled in #416.